crates/ty_project/src/metadata/options.rs

```diff
@@ -14,25 +14,57 @@ use ty_python_semantic::{ProgramSettings, PythonPath, PythonPlatform, SearchPath
 
 use super::settings::{Settings, TerminalSettings};
 
-/// The options for the project.
-#[derive(Debug, Default, Clone, PartialEq, Eq, Combine, Serialize, Deserialize)]
+#[derive(
+    Debug, Default, Clone, PartialEq, Eq, Combine, Serialize, Deserialize, OptionsMetadata,
+)]
 #[serde(rename_all = "kebab-case", deny_unknown_fields)]
 #[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
 pub struct Options {
     /// Configures the type checking environment.
+    #[option_group]
     #[serde(skip_serializing_if = "Option::is_none")]
     pub environment: Option<EnvironmentOptions>,
 
     #[serde(skip_serializing_if = "Option::is_none")]
+    #[option_group]
     pub src: Option<SrcOptions>,
 
-    /// Configures the enabled lints and their severity.
+    /// Configures the enabled rules and their severity.
+    ///
+    /// See [the rules documentation](https://github.com/astral-sh/ruff/blob/main/crates/ty/docs/rules.md) for a list of all available rules.
+    ///
+    /// Valid severities are:
+    ///
+    /// * `ignore`: Disable the rule.
+    /// * `warn`: Enable the rule and create a warning diagnostic.
+    /// * `error`: Enable the rule and create an error diagnostic.
+    ///   ty will exit with a non-zero code if any error diagnostics are emitted.
     #[serde(skip_serializing_if = "Option::is_none")]
+    #[option(
+        default = r#"{...}"#,
+        value_type = r#"dict[str, ignore | warn | error]"#,
```

Comment:
Is this following any formalized notation, or is it just meant to be some human-readable intuitive representation? The `dict[...]` and `str` parts seem like Python type notation, but in that case it should be `Literal["ignore", "warn", "error"]`, not `ignore | warn | error`.

Whatever the intended format, it's slightly odd to me that here we use an unquoted bare name both to mean "a type" (`str`) and "a string literal" (`ignore`, `warn`, `error`).

I see in some other cases below you use quoted strings with vertical bars, so maybe this is what you meant here (still not Python typing syntax quite, but a more readable variant of it):
```suggestion
        value_type = r#"dict[str, "ignore" | "warn" | "error"]"#,
```

Reply:
It's a Python-typing inspired notation but it's mostly made up (e.g. we don't use `Literal["abcd"]` because that's just too verbose).

Let me double check what notation we use for enum values in ruff